eclipse-che/che-server
# chore: Support configuring the GitHub OAuth endpoint

Signed-off-by: Igor Vinokur <ivinokur@redhat.com>



### What does this PR do?
Currently the GitHub OAuth provider is hardcoded to `https://github.com` endpoint. In order to support Github Enterprise Server, the endpoint of the GitHub OAuth provider is configurable by the oauth secret.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/21485

### How to test this PR?

1. Deploy Che from the custom `che-server` image built from the PR source: `quay.io/ivinokur/che-server:next`.
2. Apply the GitHub OAuth secret:
```YAML
kind: Secret
apiVersion: v1
metadata:
  name: github-oauth-config
  labels:
    app.kubernetes.io/component: oauth-scm-configuration
    app.kubernetes.io/part-of: che.eclipse.org
  annotations:
    che.eclipse.org/oauth-scm-server: github
    che.eclipse.org/scm-server-endpoint: 'http://37.139.177.199/'
data:
  id: YTFmOTcyYWM4ZGM4ZTRlNjJjN2I=
  secret: N2E3OGYwZjIxMmFkOGZlMGEyYjVlZGQzZDlhOTg0NmQxOGVhMDYwYg==
type: Opaque
```
The secret refers to the GitHub Enterprise Server instace, launched in my local virtualbox environment, and exposed to `http://37.139.177.199`, [see](https://www.eclipse.org/che/docs/stable/administration-guide/configuring-oauth-2-for-github/#applying-the-github-oauth-app-secret_che) for more details.
3. Ping me to start the GitHub instance :) or use your own application, but configure the endpoint in the secret.
4. Start a factory from `http://37.139.177.199/test/test.git` or your own repo.
### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
